### PR TITLE
fuzz delegation v0.5

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: fuzz/delegation/v0.5/fuzzDelegation_test.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+		err := pfe.setServiceFee(r.Intn(maxServiceFee))
+		require.Nil(t, err)
+
+		err = pfe.continueGlobalOperation()
+		require.Nil(t, err)
+
+		pfe.printServiceFeeAndDelegationCap(t)
+		pfe.printTotalStakeByType()
+	case re.WithProbability(0.05):
+		err := pfe.validateOwnerStakeShare()
```
Comment (andrei-marinica) [inline fuzz/delegation/v0.5/fuzzDelegation_test.go:221]: validations can be done after each step for extra aggressiveness

File: fuzz/delegation/v0.5/fuzzDelegation_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+		require.Nil(t, err)
+	case re.WithProbability(0.05):
+		err := pfe.validateDelegationCapInvariant()
```
Comment (andrei-marinica) [inline fuzz/delegation/v0.5/fuzzDelegation_test.go:224]: same here, validations can be done after each step for extra aggressiveness

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: fuzz/delegation/v0.5/fuzzDelegation_test.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	fuzzutil "github.com/ElrondNetwork/arwen-wasm-vm/fuzz/util"
+	mc "github.com/ElrondNetwork/elrond-vm-util/test-util/mandos/controller"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply fuzz/delegation/v0.5/fuzzDelegation_test.go:15]: Right, forgot about it :)

File: fuzz/delegation/v0.5/fuzzDelegation_test.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+	pfe, err := newFuzzDelegationExecutor(fileResolver)
+	if err != nil {
+		panic(err)
+	}
+	return pfe
+}
+
+func TestFuzzDelegation_v0_5(t *testing.T) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply fuzz/delegation/v0.5/fuzzDelegation_test.go:43]: Done.

File: fuzz/delegation/v0.5/fuzzDelegation_test.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply fuzz/delegation/v0.5/fuzzDelegation_test.go:221]: Done.

File: fuzz/delegation/v0.5/fuzzDelegation_test.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply fuzz/delegation/v0.5/fuzzDelegation_test.go:224]: Done.

**Review by andrei-marinica — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**